src/lib/field-validators/ownerFieldValidators.ts

```diff
@@ -0,0 +1,55 @@
+import { FieldValidator } from "../fieldValidator";
+
+export class AddressLine1Validator extends FieldValidator {
+  constructor() {
+    super();
+    this._rules = [
+      {
+        errorMessage: "Building number and street is a required field",
+        predicateFn: (value) => value.length === 0,
+      },
+    ];
+  }
+}
+
+export class TownOrCityValidator extends FieldValidator {
+  constructor() {
+    super();
+    this._rules = [
+      {
+        errorMessage: "Town or city is a required field",
+        predicateFn: (value) => value.length === 0,
+      },
+    ];
+  }
+}
+
+export class PostcodeValidator extends FieldValidator {
+  constructor() {
+    super();
+    this._rules = [
+      {
+        errorMessage: "Postcode is required",
+        predicateFn: (value) => value.length === 0,
+      },
+      {
+        errorMessage: "Postcode must be a valid UK postcode",
+        predicateFn: (value) => value.length > 0 && !isPostcodeValid(value),
+      },
+    ];
+  }
+}
+
+const isPostcodeValid = (postcode) => {
+  // TODO Consider call to postcode validation service, e.g. postcodes.io, here
+  // Regex matching UK postcodes is problematic
+  // https://stackoverflow.com/a/51885364
+  const postcodeRegex = /^([A-Za-z][A-Ha-hJ-Yj-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})$/;
```

Comment:
@zackads I would be tempted to leave this out for now, especially if we do not have any unit tests for this and make a note on the ticket.

I don't think we had any validation in the original ticket for this.